rfcs/text/0031-risk-fields.md

```diff
@@ -21,6 +21,18 @@ Stage X: Provide a brief explanation of why the proposal is being marked as aban
 <!--
 Stage 1: Describe at a high level how this change affects fields. Include new or updated yml field definitions for all of the essential fields in this draft. While not exhaustive, the fields documented here should be comprehensive enough to deeply evaluate the technical considerations of this change. The goal here is to validate the technical details for all essential fields and to provide a basis for adding experimental field definitions to the schema. Use GitHub code blocks with yml syntax formatting, and add them to the corresponding RFC folder.
 -->
+The `risk` fields being proposed are as follows:
+
+Field | Type | Example | Description | Use Case
+-- | -- | -- | -- | -- 
+risk.score | float | 88.73 | A score in the range 0-100 that quantifies the amount of risk associated with an entity. The higher the score, the higher the risk. | Can be used to indicate the risk associated with a particular host
```

Comment:
I suppose `risk.score` means the same as `risk_score`. I also noticed a mention of `risk_score` in the context of rules. My hope with these `risk.*` fields was to create a reusable field set that expands on the existing concept of risk score, to include things like risk level and explainability as well, and which can then be used by other assets/entities such as rules, events, hosts, users etc.

Reply:
> I also noticed a mention of risk_score in the context of rules.

Yes, `signal.rule.risk_score` is populated by the detection engine and is specified in each detection rule. While this risk scoring feature is in the early stages, it might be a good time to discuss aligning how to capture `risk` values across the Security solution.

I noted while `event.risk_score` and `signal.rule.risk_score` are mapped as `float,` the app actually won't let users define a float value:

```json
{
  "name": "Error",
  "body": {
    "statusCode": 400,
    "error": "Bad Request",
    "message": "[request body]: Invalid value \"21.5\" supplied to \"risk_score\""
  }
}
```

I don't have a strong opinion around `float` vs. `long,` but we should be consistent. I'll open an issue for Kibana asking about the `signal.rule.risk_score` behavior.

Reply:
I kicked off a [discussion](https://github.com/elastic/security-team/issues/2895) with the broader Security team regarding the concept of risk across the Security solution. My main takeaway:

> Let's focus on adding the new embeddable risk.* fields, then we can work with ECS team to determine where they apply. For example we could disallow the use of this new risk.* field set under event.* and rule.* so as to not cause confusion with event.risk_score, etc. We can worry about that subsequently.